### PR TITLE
Add dummy hash for `AbsIdealSheaf`

### PR DESCRIPTION
**Pull request by lgoettgens**
Some progress towards https://github.com/oscar-system/Oscar.jl/issues/2222.

### CODE REVIEW

File: src/AlgebraicGeometry/Schemes/Sheaves/IdealSheaves.jl [hunk 1]
```diff
@@ -587,6 +587,10 @@ function ==(I::AbsIdealSheaf, J::AbsIdealSheaf)
   return true
 end
 
+function Base.hash(I::AbsIdealSheaf, h::UInt)
+  return h
```
Comment (HechtiDerLachs) [inline src/AlgebraicGeometry/Schemes/Sheaves/IdealSheaves.jl:592]:
```suggestion
  h = hash(scheme(I), h)
  return h
```

File: src/AlgebraicGeometry/Schemes/Sheaves/IdealSheaves.jl [hunk 2]
```diff
@@ -587,6 +587,11 @@ function ==(I::AbsIdealSheaf, J::AbsIdealSheaf)
   return true
 end
 
+function Base.hash(I::AbsIdealSheaf, h::UInt)
+  h = hash(scheme(I), h)
```
Comment (afkafkafk13) [inline src/AlgebraicGeometry/Schemes/Sheaves/IdealSheaves.jl:591]: Just to be sure: You deliberately choose the hash-value of the scheme on which I is defined?

Comment (lgoettgens) [reply src/AlgebraicGeometry/Schemes/Sheaves/IdealSheaves.jl:591]: This was the suggestion of @HechtiDerLachs. I assumed that this is some property that all `==` AbsIdealSheaves have in common

Comment (afkafkafk13) [reply src/AlgebraicGeometry/Schemes/Sheaves/IdealSheaves.jl:591]: Indeed, if you were looking precisely for something that `==` AbsIdealSheaves have in common (and usually many more..), this is certainly the case. So the choice makes sense.

**Review by afkafkafk13 — APPROVED** [review]

**Pull request merged**